### PR TITLE
doc: release: 3.5: add release notes for SDHC and Disk drivers

### PR DESCRIPTION
**Pull request by danieldegrasse**
Add release notes for SDHC and disk driver updates merged since 3.4

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]

File: doc/releases/release-notes-3.5.rst [hunk 1]
```diff
@@ -475,6 +475,9 @@ Drivers and Sensors
 
 * Disk
 
+  * Ramdisk driver is now configured using devicetree, and supports multiple
+    instances
```
Comment (kartben) [inline doc/releases/release-notes-3.5.rst:479]: does this impact users to the point where entries should be added to the migration guide? Ex. Kconfig options that were removed and replaced by binding properties?

**Review by kartben — APPROVED** [review]
+1 but please see my comment re: migration guide

**Review by fabiobaltieri — APPROVED** [review]

**Comment by jhedberg** [comment]
I'll merge this. @danieldegrasse please follow up with another PR if you need to add something to the migration guide (as pointed out by @kartben in an earlier comment here).

File: doc/releases/release-notes-3.5.rst (hunk 1 above)
Comment (danieldegrasse) [reply doc/releases/release-notes-3.5.rst:479]: Yes, probably so- will make a PR for this

**Pull request merged**